pointcloud/point.go

```diff
@@ -126,11 +126,39 @@ func (bp *basicData) Value() int {
 	return bp.value
 }
 
-func (bp *basicData) SetIntensity(v uint16) Data {
-	bp.intensity = v
-	return bp
-}
+// MarshalBinary checks for the presence of color and value data. If present, it will store said data in an order byte
+// array.
+func (bp *basicData) MarshalBinary() (data []byte, err error) {
+	var dataBytes []byte
+
+	if bp.HasColor() {
+		c := color.NRGBAModel.Convert(bp.Color()).(color.NRGBA)
 
-func (bp *basicData) Intensity() uint16 {
-	return bp.intensity
+		dataBytes = append(dataBytes, c.R)
+		dataBytes = append(dataBytes, c.G)
+		dataBytes = append(dataBytes, c.B)
+		dataBytes = append(dataBytes, c.A)
+	}
+	if bp.HasValue() {
+		dataBytes = append(dataBytes, byte(bp.Value()))
```

Comment:
Current map representation [tech spec](https://docs.google.com/document/d/1es8FiYdf5XgQHrAzFLns9e2_qV202DuOsDp4rH3VSVU/edit#) requires value to be a single byte. This may limit its usefulness in pointcloud related applications. Options:
1. Require value to be a uint8
2. Increase number of bytes for value
3. Since only octree will be using this functionality for now and will always be referencing a value between 0-100, leave it as is (other potential calls for non-octree related pointcloud Data serialization may produce inaccurate results depending on value range of device